llvm/cmake/modules/GetHostTriple.cmake

```diff
@@ -2,7 +2,7 @@
 # Invokes config.guess
 
 function( get_host_triple var )
-  if( MSVC )
+  if(MSVC OR (CMAKE_HOST_SYSTEM_NAME STREQUAL Windows AND CMAKE_C_COMPILER_ID MATCHES "Clang"))
```

Comment:
This isn't about `CMAKE_HOST_SYSTEM_NAME`, it's about the target, so we should check `CMAKE_SYSTEM_NAME` instead - to make it work as expected and intended for cross compilation.

The existing check below which checks `CMAKE_HOST_SYSTEM_NAME` is for determining whether it will be possible to execute a shell script, where we do need to check the host environment.